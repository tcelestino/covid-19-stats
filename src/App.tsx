import React, { useState } from 'react';
import {
  CssBaseline,
  Container,
  AppBar,
  Link,
  Toolbar,
  Typography,
  makeStyles,
  Theme
} from '@material-ui/core';
import Countries from './components/Countries';
import CountryStats from './components/CountryStats';
import GlobalStats from './components/GlobalStats';

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh'
  },
  title: {
    fontWeight: 700,
    fontSize: '2rem'
  },
  titleText: {
    color: theme.palette.grey[50]
  },
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor: theme.palette.grey[200]
  }
}));

function App() {
  const classes = useStyles();
  const [country, setCountry] = useState<string>('');
  const handlerSelected = (countryCode: string): void => {
    setCountry(countryCode);
  };
  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar position="static" color="secondary">
        <Toolbar>
          <Typography variant="h1" className={classes.title}>
            <Link href="/" className={classes.titleText}>
              COVID-19 Stats
            </Link>
          </Typography>
        </Toolbar>
      </AppBar>
      <Container component="main" maxWidth="sm">
        <Countries onSelected={handlerSelected} />
        {!country && <GlobalStats />}
        {country && <CountryStats countryCode={country} />}
      </Container>
      <footer className={classes.footer}>
        <Container maxWidth="sm">
          <Typography variant="body1">
            Information data is supplied by{' '}
            <a
              href="https://thevirustracker.com/api"
              rel="noopener noreferrer"
              target="_blank"
            >
              Coronavirus Tracker
            </a>
          </Typography>
        </Container>
      </footer>
    </div>
  );
}

export default App;
