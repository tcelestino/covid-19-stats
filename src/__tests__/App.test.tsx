import React from 'react';
import { cleanup, render, waitForElement, fireEvent } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

describe('<App />', () => {
  // TODO: fixes warning memory leak
  it.skip('render page', async () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('should see global stats when page loaded', async () => {
    const fakeAxios = {
      // eslint-disable-next-line @typescript-eslint/camelcase
      get: jest.fn(() => jest.fn().mockResolvedValueOnce({ results: [{ total_cases: 100 }] })),
    };

    const { getByRole, container } = render(<App />);
    const loading = getByRole('progressbar');

    expect(loading).toBeInTheDocument();

    await fakeAxios.get();

    const globalStats = await waitForElement(() => container.querySelector('#global-stats'), { timeout: 6000 });
    expect(loading).not.toBeInTheDocument();
    expect(globalStats).toHaveTextContent('Global Stats');
    expect(fakeAxios.get).toHaveBeenCalledTimes(1);
  });

  it('should see country stats when select a country on select', async () => {
    const { getByRole, debug } = render(<App />);

    const listCountries = getByRole('button') as HTMLDivElement;
    fireEvent.click(listCountries);
    debug();
  });
});
