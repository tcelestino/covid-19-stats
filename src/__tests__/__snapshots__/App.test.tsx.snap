// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorSecondary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h1
          class="MuiTypography-root makeStyles-title-2 MuiTypography-h1"
        >
          Covid-19 Stats
        </h1>
      </div>
    </header>
    <main
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-85 MuiInputBase-fullWidth MuiInput-fullWidth"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby=" "
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          <span>
            â€‹
          </span>
        </div>
        <input
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <div
        class="makeStyles-root-142"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
      
    </main>
    <footer
      class="makeStyles-footer-3"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Information data is supplied by
           
          <a
            href="https://thevirustracker.com/api"
            rel="noopener noreferrer"
            target="_blank"
          >
            Coronavirus Tracker
          </a>
        </p>
      </div>
    </footer>
  </div>
</div>
`;
